hugolib/page.go

```diff
@@ -937,5 +953,12 @@ func (p *Page) TargetPath() (outfile string) {
 		outfile = helpers.ReplaceExtension(p.Source.LogicalName(), p.Extension())
 	}
 
-	return filepath.Join(strings.ToLower(helpers.MakePath(p.Source.Dir())), strings.TrimSpace(outfile))
+	return p.addLangPrefix(filepath.Join(strings.ToLower(helpers.MakePath(p.Source.Dir())), strings.TrimSpace(outfile)))
+}
+
+func (p *Page) addLangPrefix(outfile string) string {
+	if p.lang == "" {
+		return outfile
+	}
+	return p.lang + "/" + outfile
```

Comment:
On line 940, we take a url-style path and converts it to an OS dependent path with `filepath.FromSlash`.  `filepath.FromSlash` will convert `posts/index.html` to `posts\index.html` on Windows.

@bep is correct.  In this context, you want to use `filepath.Join`.